internal/terraform/eval_conditions.go

```diff
@@ -91,24 +99,46 @@ func evalCheckRules(typ checkType, rules []*configs.CheckRule, ctx EvalContext,
 			diags = diags.Append(&hcl.Diagnostic{
 				Severity:    hcl.DiagError,
 				Summary:     errInvalidCondition,
-				Detail:      fmt.Sprintf("Invalid validation condition result value: %s.", tfdiags.FormatError(err)),
+				Detail:      fmt.Sprintf("Invalid condition result value: %s.", tfdiags.FormatError(err)),
 				Subject:     rule.Condition.Range().Ptr(),
 				Expression:  rule.Condition,
 				EvalContext: hclCtx,
 			})
 			continue
 		}
 
-		if result.False() {
-			diags = diags.Append(&hcl.Diagnostic{
-				Severity:    hcl.DiagError,
-				Summary:     typ.FailureSummary(),
-				Detail:      rule.ErrorMessage,
-				Subject:     rule.Condition.Range().Ptr(),
-				Expression:  rule.Condition,
-				EvalContext: hclCtx,
-			})
+		if result.True() {
+			continue
+		}
+
+		var errorMessage string
+		if !errorDiags.HasErrors() && errorValue.IsKnown() && !errorValue.IsNull() {
+			var err error
+			errorValue, err = convert.Convert(errorValue, cty.String)
+			if err != nil {
+				diags = diags.Append(&hcl.Diagnostic{
+					Severity:    hcl.DiagError,
+					Summary:     "Invalid error message",
+					Detail:      fmt.Sprintf("Error message must be a string: %s.", tfdiags.FormatError(err)),
```

Comment:
If I'm understanding this flow correctly, I think a typical error message from `convert.Convert` here would be "a string is required", making this full message be "Error message must be a string: a string is required", which seems redundant.

I think in similar situations elsewhere we've led with something like "Unsuitable value for error message: ", so that the full message would be a like "Unsuitable value for error message: a string is required."